data-analytics/dataflow-python-examples/dataflow_python_examples/data_lake_to_mart_cogroupbykey.py

```diff
@@ -0,0 +1,178 @@
+            traceback.print_exc()
+            logging.error("Account Not Found error: %s", err)
+
+        for order in result:
+            order.update(account_details)
+
+        return result
+
+
+def run(argv=None):
+    """The main function which creates the pipeline and runs it."""
+    parser = argparse.ArgumentParser()
+    # Here we add some specific command line arguments we expect.
+    # This defaults the output table in your BigQuery you'll have
+    # to create the example_data dataset yourself using bq mk temp
+    parser.add_argument('--output', dest='output', required=False,
+                        help='Output BQ table to write results to.',
+                        default='lake.orders_denormalized_cogroupbykey')
+
+    # Parse arguments from the command line.
+    known_args, pipeline_args = parser.parse_known_args(argv)
+
+    # DataLakeToDataMartCGBK is a class we built in this script to hold the logic for
+    # transforming the file into a BigQuery table.  It also contains an example of
+    # using CoGroupByKey
+    data_lake_to_data_mart = DataLakeToDataMartCGBK()
+
+    schema = parse_table_schema_from_json(data_lake_to_data_mart.schema_str)
+    pipeline = beam.Pipeline(options=PipelineOptions(pipeline_args))
+
+    # This query returns details about the account, normalized into a
+    # different table.  We will be joining the data in to the main orders dataset in order
+    # to create a denormalized table.
+    account_details_source = (
+        pipeline
+        | 'Read Account Details from BigQuery ' >> beam.io.Read(
+            beam.io.BigQuerySource(query="""
+                SELECT
+                  acct_number,
+                  acct_company_name,
+                  acct_group_name,
+                  acct_name,
+                  acct_org_name,
+                  address,
+                  city,
+                  state,
+                  zip_code,
+                  country
+                FROM
+                  `python-dataflow-example.example_data.account`
+            """, use_standard_sql=True))
+        # This next stage of the pipeline maps the acct_number to a single row of
+        # results from BigQuery.  Mapping this way helps Dataflow move your data arround
+        # to different workers.  When later stages of the pipeline run, all results from
+        # a given account number will run on one worker.
+        | 'Map Account to Order Details' >> beam.Map(
+            lambda row: (
+                row['acct_number'], row
+            )))
+
+    # Read the orders from BigQuery.  This is the source of the pipeline.  All further
+    # processing starts with rows read from the query results here.
+    orders = (
+        pipeline
+        | 'Read Orders from BigQuery ' >> beam.io.Read(
+            beam.io.BigQuerySource(query="""
+                                    SELECT
+                               -- select * is not recommended in production code.  For brevity, 
```

Comment:
You're writing an example, so write a good example. "Do as I do" is better than "do as I say".  If it's less verbose to write a bit of Python that does a ```', '.join(field_names)``` to create the field list part of the query, then do that.